### PR TITLE
GraphQL caching

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This PR adds documentation for https://github.com/magento/graphql-ce/issues/230 and internal ticket MC-15959

### CODE REVIEW

**Comment by keharper** [comment]
This PR replaces #4260

File: guides/v2.3/graphql/caching.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* `cmsPage`
+* `products`
+* `urlResolver`
+
+Magento explicitly disallows caching the following queries.
+
+* `cart`
+* `country`
+* `countries`
+* `currency`
```
Comment (osrecio) [inline guides/v2.3/graphql/caching.md]: Just out of curiosity, Why wouldn't it be valid to cache  countries, country and currency?

File: guides/v2.3/graphql/develop/create-graphqls-file.md [hunk 2]
```diff
@@ -155,3 +114,21 @@ type Query {
         identifier: String @doc(description: "Identifier of the CMS page")
 ...
 ```
+
+## Query caching
+
+The `@cache` directive defines whether the results of certain queries can be cached. Queries relating to products, categories, and CMS may be cached.
+
+Define cachable queries in the following manner:
```
Comment (osrecio) [inline guides/v2.3/graphql/develop/create-graphqls-file.md]: cachable or cacheable? Check the line 130:  `@cache(cacheable: false)`

**Comment by osrecio** [comment]
Hi @keharper , I dropped a couple of comments :)

File: guides/v2.3/graphql/caching.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    # For multi site configurations to not cache each other's content
+    if (req.http.host) {
+        hash_data(req.http.host);
+    } else {
+        hash_data(server.ip);
+    }
+
+    if (req.url ~ "graphql") {
```
Comment (prabhuram93) [inline guides/v2.3/graphql/caching.md]: The condition should be if (req.url ~ "/graphql"), it was updated in the code.

Comment (keharper) [reply guides/v2.3/graphql/caching.md]: Fixed

File: guides/v2.3/graphql/send-request.md [hunk 4]
```diff
@@ -23,12 +21,15 @@ The previous example is equivalent to the following query. You could send the qu
...
+Some queries sent as a GET request can be cached. See [GraphQL caching]({{page.baseurl}}/graphql/caching.html) for more information.
+
 ## Request headers
 
 Magento accepts the following headers in a GraphQL request:
 
 Header name | Value | Description
 --- | --- | ---
 `Authorization` | `Bearer <authorization_token>` | A customer token. [Get customer authorization token]({{ page.baseurl }}/graphql/get-customer-authorization-token.html) describes how to generate the token.
+`Content-Currency` | A valid currency code, such as `USD` | This header is required only if the currency is not the store view's default currency.
```
[the remark this answers is not in the document]
Comment (keharper) [reply guides/v2.3/graphql/send-request.md]: Created internal ticket MC-16230 to track the doc work when the code changes.

**Pull request merged**